### PR TITLE
Reviewed and updated 16 files of the public API

### PR DESCRIPTION
**Pull request by HalynaDumych**
Modified the existing comments and added the missing comments to
the public APIs of the following libs:
* olp-sdk-authentication/lib/
* olp-sdk-dataservice-read/lib/
* olp-sdk-dataservice-read/lib/cache/
* olp-sdk-dataservice-read/lib/utils/

Relates-to: OLPEDGE-869

Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts [hunk 1]
```diff
@@ -49,25 +49,23 @@ class DeferredPromise<T> {
 }
 
 /**
- * `DataStoreDownloadManager` for downloading URLs.
+ * Downloads URLs.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts]: not URLs, This is an implementation of DownloadManager interface requested by datastore-api module for fetching data from the backend or sending requests to the backend. This implementation allows us  to fetch blobs or JSONs data with aborting or re-sending (on timeouts) mechanisms with limiting for maximum parallel requests

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts [hunk 2]
```diff
@@ -46,6 +46,16 @@ async function addBearerToken(
 }
 
 export class DataStoreRequestBuilder extends RequestBuilder {
+    /**
+     * Cerates the `DataStoreRequestBuilder` instance.
+     * @param downloadManager Downloads data from the specified URL.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: not download data. It's a configured instance of the implementation DownloadManager interface. User can use our default DatastoreDownloadManager or create own implementation

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts [hunk 3]
```diff
@@ -46,6 +46,16 @@ async function addBearerToken(
 }
 
 export class DataStoreRequestBuilder extends RequestBuilder {
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: Please add a comment about the class. It's an implementation of  RequestBuilder, uses to configure the instance with base-url to the service and implementation of downloadBlob and download methods

File: @here/olp-sdk-dataservice-read/lib/DownloadManager.ts [hunk 4]
```diff
@@ -18,22 +18,20 @@
...
- *
- * Interface for DataStoreRequestBuilder.
- *
- * DataStoreRequestBuilder has own default implementation of download manager.
- *
- * If you want to use your own download manager, you are able implement this interface end set it.
+ * Downloads URLs.
+ * An interface for `DataStoreRequestBuilder`.
+ * `DataStoreRequestBuilder` has its own default implementation of the download manager.
+ * If you want to use your download manager, you can implement this interface, and then set your download manager.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/DownloadManager.ts]: it's not for Downloads URLs and not an interface for `DataStoreRequestBuilder`. It's an interface for @here/olp-sdk-dataservice-api to be able to send requests to the services

File: @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts [hunk 5]
```diff
@@ -20,9 +20,7 @@
 import { HRN } from "../HRN";
 import { KeyValueCache } from "./KeyValueCache";
 
-/**
- * The list of API names of endpoints exists in the API
- */
+// The list of API endpoints to which the Open Location Platform (OLP) SDK for Typescript has access.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts]: The list names of API endpoints to which the Open Location Platform (OLP) SDK for Typescript uses.

File: @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts [hunk 6]
```diff
@@ -65,10 +69,10 @@ export class ApiCacheRepository {
     }
 
     /**
-     * Store key,value pair into the cache
-     * @param service
-     * @param serviceVersion
-     * @returns The base url of the service or undefined if not exists
+     * Gets a base URL from the cache.
+     * @param service TThe name of the API service for which you want to get the base URL.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts]: TThe

File: @here/olp-sdk-dataservice-read/lib/cache/KeyValueCache.ts [hunk 7]
```diff
@@ -18,40 +18,42 @@
  */
 
 /**
- * Class for cache expecting a key,value pair.
+ * Used to provide the `KeyValueCache` instance to the `ApiCacheRepository` instance.
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/cache/KeyValueCache.ts]: not just to the `ApiCacheRepository` instance. Later we'll have more instances for more caching items (configuration, statistics, etc...) It's a base in-memory caching instance and all *Repository  will use it for reading or caching information

File: @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts (hunk 1 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts (hunk 2 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts (hunk 3 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/DownloadManager.ts (hunk 4 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/DownloadManager.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts (hunk 5 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts (hunk 6 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/cache/KeyValueCache.ts (hunk 7 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/cache/KeyValueCache.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts (hunk 2 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts (hunk 1 above)
Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts]: Fixed

File: @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts (hunk 6 above)
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts]: I still see "TThe"

Comment (HalynaDumych) [reply @here/olp-sdk-dataservice-read/lib/cache/ApiCacheRepository.ts]: Sorry
Fixed

**Pull request merged**